addons/im_livechat/static/src/legacy/public_livechat_chatbot.js

```diff
@@ -0,0 +1,882 @@
+                this._chatbotAddMessage(result.chatbot_posted_message);
+            }
+
+            this._chatbotCurrentStep = result.chatbot_step;
+
+            this._chatbotProcessStep();
+        } else {
+            // did not find next step -> end the script
+            this._chatbotCurrentStep['chatbot_step_is_last'] = true;
+            this._renderMessages();
+            this._chatbotEndScript();
+        }
+
+        this._chatbotSaveSession();
+
+        return result;
+    },
+
+     //--------------------------------------------------------------------------
+     // Private - LiveChat Overrides
+     //--------------------------------------------------------------------------
+
+    /**
+     *
+     * @override
+     */
+     _askFeedback: function () {
+        this._super(...arguments);
+
+        this._chatWindow.$('.o_livechat_chatbot_end').addClass('d-none');
+        this._chatWindow.$('.o_composer_text_field')
+            .removeClass('d-none')
+            .val('');
+     },
+
+    /**
+     * Small override:
+     * If we resume an existing chatbot script, check after opening the chat window if we are on
+     * the last step, and "end script" if so.
+     *
+     * @private
+     * @override
+     */
+    _openChatWindow: function () {
+        return this._super(...arguments).then(() => {
+            window.addEventListener('resize', () => {
+                if (this._chatWindow) {
+                    this._chatWindow.scrollToBottom();
+                }
+            });
+
+            if (this._chatbotCurrentStep && this._messages && this._messages.length !== 0) {
+                this._chatbotProcessStep();
+            }
+        });
+    },
+    /**
+     * @private
+     * @override
+     */
+    _prepareGetSessionParameters: function () {
+        const parameters = this._super(...arguments);
+
+        if (this._isChatbot) {
+            parameters.chatbot_script_id = this._chatbot.chatbot_script_id;
+        }
+
+        return parameters;
+    },
+    /**
+     * @private
+     */
+    _renderMessages: function () {
+        this._super(...arguments);
+
+        var self = this;
+
+        this._chatWindow.$('.o_thread_message:last .o_livechat_chatbot_options li').each(function () {
+            $(this).on('click', self._onChatbotOptionClicked.bind(self));
+        });
+
+        this._chatWindow.$('.o_livechat_chatbot_main_restart').on('click',
+            this._onChatbotRestartScript.bind(this));
+
+        if (this._messages.length !== 0) {
+            const lastMessage = this._messages[this._messages.length - 1];
+            const stepAnswers = lastMessage.getChatbotStepAnswers();
+            if (stepAnswers && stepAnswers.length !== 0 && !lastMessage.getChatbotStepAnswerId()) {
+                this._chatbotDisableInput(_t('Select an option above'));
+            }
+        }
+    },
+    /**
+     * When the website visitor sends a message, we need to act depending on our current state:
```

Comment:
Note that using "website visitor" is a bit confusing, as we don't have that model used in livechat as far as I remember. Customer ?